corehq/apps/hqadmin/tests/test_clone_domain.py

```diff
@@ -92,12 +92,23 @@ def locations_snapshot(domain):
         # Make sure the locations are only related to locations in the same domain
         for domain in (self.old_domain, self.new_domain):
             locs_in_domain = SQLLocation.objects.filter(domain=domain)
-            related_locs = (SQLLocation.objects.get_queryset_descendants(locs_in_domain, include_self=True)
-                            | SQLLocation.objects.get_queryset_ancestors(locs_in_domain, include_self=True))
-            self.assertItemsEqual(
-                related_locs.order_by('domain').distinct('domain').values_list('domain', flat=True),
-                [domain]
+            related_to_other_domain = (
+                SQLLocation.objects
+                .get_queryset_ancestors(locs_in_domain)
+                # exclude here instead of at end (on union queryset) because
+                # django discards filters on union queries?? (django bug?)
```

Comment:
That seems like a frustrating thing to figure out. Any concerns this impacts real code?

Reply:
Great question. I imagine we would need to evaluate every use of `QuerySet.union(...)` to determine that. After a cursory look, I think this may be the only place it's used.